src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -359,9 +359,14 @@ private function getDenormalizer(mixed $data, string $class, ?string $format, ar
 
                 $supportedTypes = $normalizer->getSupportedTypes($format);
 
+                $doesClassRepresentCollection = str_ends_with($class, '[]');
+
                 foreach ($supportedTypes as $supportedType => $isCacheable) {
+                    $doesSupportedTypeRepresentCollection = str_ends_with($supportedType, '[]');
```

Comment:
Maybe I would avoid using so long vairable names and in the matter of fact I think that this variable is not needed here as is used only once.

Reply:
I'm fine with changing this or leaving the variable as-is just due to readability - I'm happy to go either way the SF maintainers want.

Reply:
Sure, just my personal opinion here, not fault at all. :)

Reply:
inlined